src/context/box-repository.ts

```diff
@@ -158,4 +158,31 @@ export class BoxRepository extends BaseRepository<BoxEntity> {
 
     return globalIndex;
   }
+
+  public async getAddressesBoxCount(options: {
+    where: { addresses: string[]; maxHeight?: number };
+  }) {
+    let baseQuery = this.repository
+      .createQueryBuilder("box")
+      .select("COUNT(*)", "boxesCount")
+      .addSelect("box.address", "address")
+      .where("box.address IN (:...addresses)")
+      .groupBy("box.address")
+      .setParameters(
+        removeUndefined({
+          addresses: options.where.addresses,
+          height: options.where.maxHeight
+        })
+      );
```

Comment:
For improved code semantics, let's set params after `maxHeight` condition and near to return instruction.